src/selectors/layout.js

```diff
@@ -17,9 +18,9 @@ const getVisibleSidebar = state => state.visible.sidebar;
  * which don't affect layout.
  */
 export const getGraph = createSelector(
-  [getVisibleNodes, getVisibleEdges, getHasVisibleLayers],
-  (nodes, edges, hasVisibleLayers) => {
-    if (!nodes.length || !edges.length) {
+  [getVisibleNodes, getVisibleEdges, getHasVisibleLayers, getFontLoaded],
+  (nodes, edges, hasVisibleLayers, fontLoaded) => {
+    if (!fontLoaded || !nodes.length || !edges.length) {
```

Comment:
Can the check on `fontLoaded` be pushed higher up? I know it's an optimisation, but as it doesn't seem to be this function's responsibility to know about this normally, so it would be cleaner to move it.

Reply:
I hear you, and I'm not 100% happy with this approach. However, I previously had this check way higher up elsewhere, and then I made a change which called this function elsewhere, and then it started being called prematurely again. So I think the most sensible thing to do is to put the check right here (in addition to some of the higher up functions), so that it's _explicit_ about not being called, and won't be accidentally affected by a future change.

Reply:
Yeah I suspected it might be something like this, it's not ideal but if that's the most practical approach then seems fine with me, thanks for confirming.